src/executor/stack/executor.rs

```diff
@@ -404,7 +404,6 @@ impl<'config, 'precompiles, S: StackState<'config>, P: PrecompileSet>
 			// EIP-3860
 			if init_code.len() > limit {
 				self.state.metadata_mut().gasometer.fail();
-				let _ = self.exit_substate(StackExitKind::Failed);
 				return Err(ExitError::OutOfGas);
```

Comment:
Given from spec: "Note that this is similar to transactions considered invalid for not meeting the intrinsic gas cost requirement"

`return Err(ExitError::OutOfGas);` doesn't appear to be the correct `ExitError`

Reply:
Good point, fixed that too.